whittle/metrics/mag.py

```diff
@@ -0,0 +1,45 @@
+import torch
+
+from whittle.models.gpt import GPT
+
+
+def weight_magnitude(model: GPT):
+    magnitude = 0
+    magnitude += weight_magnitude_linear_layer(model.lm_head)
+    magnitude += weight_magnitude_embedding(model.transformer.wte)
+    for block in model.transformer.h:
+        magnitude += weight_magnitude_layer_norm(block.norm_1)
+        magnitude += weight_magnitude_attention(block.attn)
+        magnitude += weight_magnitude_linear_layer(block.mlp.fc)
```

Comment:
This will fail for llama, mistral and gemma. We should again handle the three MLP types separately